dataprofiler/labelers/base_model.py

```diff
@@ -21,7 +21,7 @@ def __new__(
             cls, clsname, bases, attrs
         )
         new_class._register_subclass()
-        return new_class
+        return cast(AutoSubRegistrationMeta, new_class)
```

Comment:
This might be better to do a `T` generic because really the class should be original of the input? Otherwise, would this convert something using this metaclass to that type later in code?

Is that why loading has the issue as well?

Reply:
The issue is that `new_class` needs to be typed as `Any` initially because otherwise mypy complains that `._register_subclass()` doesn't exist. I'm thinking we could also do:
```py
new_class: AutoSubRegistrationMeta = super(AutoSubRegistrationMeta, cls).__new__(
    cls, clsname, bases, attrs
)
new_class._register_subclass() # type: ignore
return new_class
```
which does pass mypy.

Reply:
I agree that`new_class: AutoSubRegistrationMeta = ...` would not work either. However, I was wondering if there's a generic approach that could fix this that allows us to do `new_class: T = ...` or some sort where `T` is the `cls` or something of the like. If it is too complex, fair. Either way, it doesn't feel right to me that `cast(AutoSubRegistrationMeta, new_class)` shouldn't be some `cast(T, new_class)` as the output is not a class `AutoSubRegistrationMeta`, right?

Reply:
```py
T = TypeVar("T", bound="AutoSubRegistrationMeta")


class AutoSubRegistrationMeta(abc.ABCMeta):
    """For registering subclasses."""

    def __new__(
        cls: Type[T], clsname: str, bases: Tuple[type, ...], attrs: Dict[str, object]
    ) -> T:
        """Create auto registration object and return new class."""
        new_class: T = super(AutoSubRegistrationMeta, cls).__new__(
            cls, clsname, bases, attrs
        )
        new_class._register_subclass()  # type: ignore
        return new_class
```
Was able to get it working with this change (which I just committed). Couldn't get around the `._register_subclass()` line because that's done dynamically.